internal/dao/pod.go

```diff
@@ -394,25 +428,32 @@ func readLogs(ctx context.Context, wg *sync.WaitGroup, stream io.ReadCloser, out
 		var item *LogItem
 		if bytes, err := r.ReadBytes('\n'); err == nil {
 			item = opts.ToLogItem(tview.EscapeBytes(bytes))
+		} else if errors.Is(err, io.EOF) {
+			e := fmt.Errorf("stream closed %w for %s", err, opts.Info())
+			item = opts.ToErrLogItem(e)
+			slog.Warn("Log reader EOF",
+				slogs.Container, opts.Info(),
+				slogs.Error, e,
+			)
+			// For EOF, we want to retry the connection
+			return
 		} else {
-			if errors.Is(err, io.EOF) {
-				e := fmt.Errorf("stream closed %w for %s", err, opts.Info())
-				item = opts.ToErrLogItem(e)
```

Comment:
in both case the log item is created but never used... There is no way for the call site to distinguish EOF vs stream error. It's not clear in this PR what happens to the stream when the credentials expired? Is it closed or does it error out??
Also streamDone chan does not provide any real convenience. Better to know if this warrants a retry or we need to bail out. Thus we can't determine a legit close ie init container vs main container crapping out.
We need to be very careful here since we are in a tight loop and firing off gors. 
The call site needs to get a clear indication of what action to take next aka cancel, retry or bail out.
IMHO only stream errors should warrant retries.